src/main/java/teammates/storage/sqlapi/AccountRequestsDb.java

```diff
@@ -129,6 +131,17 @@ public AccountRequest updateAccountRequest(AccountRequest accountRequest)
     public void deleteAccountRequest(AccountRequest accountRequest) {
         if (accountRequest != null) {
             delete(accountRequest);
+            deleteDocumentByAccountRequestId(accountRequest.getId());
+        }
+    }
+
+    /**
+     * Removes search document for the given account request.
+     */
+    public void deleteDocumentByAccountRequestId(UUID accountRequestId) {
+        if (getSearchManager() != null) {
+            getSearchManager().deleteDocuments(
+                    Collections.singletonList("java.util.UUID:" + accountRequestId.toString()));
```

Comment:
nit: let's add a comment here explaining that id is saved as "java.util.UUID:" in solr